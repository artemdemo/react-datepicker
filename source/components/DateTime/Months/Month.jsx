import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import { propIsMoment } from '../propTypes';

import './Month.less';

class Month extends React.Component {
    clickHandler(validDate) {
        const { date, onClick } = this.props;
        if (validDate) {
            onClick && onClick(date);
        }
    }

    render() {
        const { date, faded, isValidDate } = this.props;
        const monthClass = classnames({
            'datetime-month': true,
        });
        const validDate = isValidDate ? isValidDate(date) : true;
        const contentClass = classnames({
            'datetime-month__content': true,
            'datetime-month__content_faded': faded,
            'datetime-month__content_disabled': !validDate,
        });
        return (
            <div
                className={monthClass}
                onClick={this.clickHandler.bind(this, validDate)}
            >
                <div className={contentClass}>
                    {date.format('MMM')}
                </div>
            </div>
        );
    }
}

Month.displayName = 'Month';

Month.propTypes = {
    date: propIsMoment.isRequired,
    onClick: PropTypes.func,
    isValidDate: PropTypes.func,
    faded: PropTypes.bool,
};

Month.defaultProps = {
    onClick: null,
    isValidDate: null,
    faded: false,
};

export default Month;
