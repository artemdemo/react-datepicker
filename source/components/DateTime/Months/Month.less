@import '../styles/general';

.datetime-month {
    float: left;
    width: 25%;
    height: 67px;
    text-align: center;
    display: table;
}

    .datetime-month__content {
        .clickable();
        display: table-cell;
        vertical-align: middle;

        border: 2px solid transparent;
        background-color: transparent;
        height: 100%;
    }
        .datetime-month__content_faded {
            color: fade(@text-color, 35%);
        }
        .datetime-month__content_disabled {
            color: fade(@text-color, 15%);
            cursor: not-allowed;

            &:hover {
                background: transparent;
            }
        }
