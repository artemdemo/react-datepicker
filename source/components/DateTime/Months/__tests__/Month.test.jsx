import React from 'react';
import moment from 'moment';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import Month from '../Month';

describe('<Month />', () => {
    const date = moment(1502459094000);

    it('simple render', () => {
        const tree = renderer.create(
            <Month date={date} />,
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('onClick', () => {
        let clicked = false;
        const wrapper = mount(
            <Month
                date={date}
                onClick={() => clicked = true}
            />,
        );
        wrapper.find('.datetime-month').first().simulate('click');
        expect(clicked).toEqual(true);
    });

    it('isValidDate', () => {
        const tree = renderer.create(
            <Month
                date={date}
                isValidDate={() => false}
            />,
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('faded', () => {
        const tree = renderer.create(
            <Month
                date={date}
                faded
            />,
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
