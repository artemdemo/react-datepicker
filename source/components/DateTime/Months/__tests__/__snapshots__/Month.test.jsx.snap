// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Month /> faded 1`] = `
<div
  className="datetime-month"
  onClick={[Function]}
>
  <div
    className="datetime-month__content datetime-month__content_faded"
  >
    Aug
  </div>
</div>
`;

exports[`<Month /> isValidDate 1`] = `
<div
  className="datetime-month"
  onClick={[Function]}
>
  <div
    className="datetime-month__content datetime-month__content_disabled"
  >
    Aug
  </div>
</div>
`;

exports[`<Month /> simple render 1`] = `
<div
  className="datetime-month"
  onClick={[Function]}
>
  <div
    className="datetime-month__content"
  >
    Aug
  </div>
</div>
`;
